go/simple/v2/ampli/ampli.go

```diff
@@ -0,0 +1,388 @@
+
+		return
+	}
+
+	if options.Client.isEmpty() {
+		options.Client = LoadClientOptions{
+			Configuration: DefaultConfiguration(),
+		}
+	}
+
+	var apiKey string
+	if options.Client.APIKey != "" {
+		apiKey = options.Client.APIKey
+	} else if options.Environment != "" {
+		apiKey = APIKey[options.Environment]
+	} else if options.Client.Configuration.APIKey != "" {
+		apiKey = options.Client.Configuration.APIKey
+	}
+
+	if !(apiKey != "" || options.Client.Instance != nil) {
+		log.Default().Printf("Error: Ampli.Load() requires option.Environment, " +
+			"and apiKey from either options.Instance.APIKey or APIKey[options.Environment], " +
+			"or options.Instance.Instance")
+	}
+
+	var configuration amplitude.Config
+	if !options.Client.Configuration.IsEmpty() {
+		configuration = options.Client.Configuration
+	} else {
+		configuration = DefaultConfiguration()
+	}
+
+	if configuration.Plan == (amplitude.Plan{}) {
+		configuration.Plan = DefaultConfiguration().Plan
+	}
+
+	if options.Client.Instance != nil {
+		a.Client = options.Client.Instance
+	} else {
+		configuration.APIKey = apiKey
+		a.Client = amplitude.NewClient(configuration)
+	}
+}
+
+// InitializedAndEnabled checks if Ampli is initialized and enabled.
+func (a *Ampli) InitializedAndEnabled() bool {
+	a.mutex.Lock()
+	defer a.mutex.Unlock()
+	return !a.Disabled && a.Client != nil
+}
+
+func (a *Ampli) setUserID(userID string, eventOptions *amplitude.EventOptions) {
+	if userID != "" {
+		eventOptions.UserID = userID
+	}
+}
+
+// Track tracks an StronglyTypedEvent.
+func (a *Ampli) Track(userID string, event StronglyTypedEvent, eventOptions amplitude.EventOptions) {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	a.setUserID(userID, &eventOptions)
+
+	baseEvent := event.ToEvent()
+	baseEvent.EventOptions = eventOptions
+
+	a.Client.Track(baseEvent)
+}
+
+// Identify identifies a user and set user properties.
+func (a *Ampli) Identify(userID string, identify *Identify, eventOptions amplitude.EventOptions) {
+	a.Track(userID, identify, eventOptions)
+}
+
+// GroupIdentify identifies a group and set group properties.
+func (a *Ampli) GroupIdentify(groupType string, groupName string, group *Group, eventOptions amplitude.EventOptions) {
+	group.groups = map[string][]string{groupType: {groupName}}
+	event := group.ToEvent()
+	event.EventOptions = eventOptions
+
+	a.Client.Track(event)
+}
+
+// SetGroup sets group for the current user.
+func (a *Ampli) SetGroup(userID string, groupType string, groupName []string, eventOptions amplitude.EventOptions) {
+	a.setUserID(userID, &eventOptions)
+
+	a.Client.SetGroup(groupType, groupName, eventOptions)
+}
+
+// Flush flushes events waiting in buffer.
+func (a *Ampli) Flush() {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	a.Client.Flush()
+}
+
+// Shutdown disables and shutdowns Ampli Instance.
+func (a *Ampli) Shutdown() {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	a.Client.Shutdown()
```

Comment:
Should be the last line, I think. After disabling.

Reply:
Correct. Current sequence may still be call track even when client is shutdown.

I should also load first, then set `Disabled` to prevent the situation where track is called but client is not ready  

Updated both